### ISSUE TITLE
react-input-autosize "window" reference breaks test environment

### ISSUE THREAD

**Issue by sheminusminus** [JedWatson/react-select#2270]
## Are you reporting a bug or runtime error?
Yes.

## Version
Any version of `react-select` that pulls in `react-input-autosize` at `2.2.0`. 

## Summary
Our test suite will no longer run (in a headless environment) due to this error:
```
/node_modules/react-input-autosize/lib/AutosizeInput.js:57
var isIE = typeof window === 'undefined' && window.document ? false : /MSIE |Trident\/|Edge\//.test(window.navigator.userAgent);
                                            ^

ReferenceError: window is not defined
    at Object.<anonymous> (/Users/emily/webapp/node_modules/react-input-autosize/lib/AutosizeInput.js:57:45)
```

**Comment by sheminusminus** [comment]
Original version number I posted was incorrect. Should have been react-input-autosize at 2.2.0. Leaving this comment in case it triggers an email or something.

**Comment by hafeez1042** [comment]
i think [this](https://github.com/JedWatson/react-input-autosize/issues/123) is the issue.

**Comment by gburnes** [comment]
Is there an easy way to get around this? This is causing build failures in my production instances and will most definitely become a P0 issue if we can't find a solution.

**Comment by davidkwan95** [comment]
It should be fixed with [this PR](https://github.com/JedWatson/react-input-autosize/pull/124#issuecomment-356445830)

**Comment by hafeez1042** [comment]
The issue is fixed on [react-input-autosize](https://github.com/JedWatson/react-input-autosize), hop it will work.

**Comment by JedWatson** [comment]
Fixed in react-input-autosize@2.2.1
